FileDescriptorMeter.c

```diff
@@ -0,0 +1,80 @@
+/*
+htop - PressureStallMeter.c
+(C) 2022 htop dev team
+Released under the GNU GPLv2+, see the COPYING file
+in the source distribution for its full text.
+*/
+
+#include "FileDescriptorMeter.h"
+
+#include <math.h>
+#include <stdbool.h>
+#include <string.h>
+
+#include "CRT.h"
+#include "Meter.h"
+#include "Object.h"
+#include "Platform.h"
+#include "RichString.h"
+#include "XUtils.h"
+
+
+static const int FileDescriptorMeter_attributes[] = {
+   FILE_DESCRIPTOR_USED,
+   FILE_DESCRIPTOR_MAX
+};
+
+static void FileDescriptorMeter_updateValues(Meter* this) {
+   this->values[0] = 0;
+   this->values[1] = 1;
+
+   Platform_getFileDescriptors(&this->values[0], &this->values[1]);
```

Comment:
why not set `FileDescriptorMeter_class.maxItems = 1,` and pass `&this->total` as second argument?

Reply:
Intentional, as there are 2 values internally, but the bar meter is just to show one of them. Basically adapted from the PSI meters.

Reply:
What is the sensible `this->total` value for this meter?

Reply:
As it is set in the source: The maximum available file descriptors as reported by the system. This just so happens to also be the second value reported, hence that one assignment of `this->total = this->values[1];`.